test/benchmarks/benchmark.py

```diff
@@ -343,35 +343,56 @@ def main(args):
         "'python3 1gb_pytorch.py'",
     ]
 
-    # run in a loop
-    num_samples = 5
-    for x in range(len(jobs)):
-        print("Starting benchmarks for job {} with command {}".format(jobs[x], cmds[x]))
-        job = jobs[x]
-        for y in range(num_samples):
-            jobID = job + "-" + str(y)
-            process_stats = run_exec(cmds[x], jobID)
-            # wait a few seconds for memory to allocate
-            time.sleep(5)
-
-            # we don't mutate jobID for checkpoint/restore here so we can pass the unadulterated one to our csv
-            run_checkpoint(daemon_pid, jobID, output_dir, process_stats)
-            time.sleep(3)
-
-            run_restore(daemon_pid, jobID, output_dir)
-            time.sleep(3)
-
-            terminate_process(process_stats["pid"])
-
-    if local:
-        return
-    # unique uuid for blob id
-    blob_id = "benchmark-data-" + str(time.time())
-    push_otel_to_bucket("/cedana/data.json", blob_id)
-    attach_bucket_id("benchmark_output.csv", blob_id)
-    push_to_bigquery()
-    # delete benchmarking folder
-    cleanup()
+    if "--correctness" in args:
+        job = "nn-1gb"
+        cmd = "'python3 1gb_pytorch_correctness.py'"
+        print("Starting correctness test for job {} with command {}".format(job, cmd))
+        jobID_base = job + "-base"
+        jobID_saved = job + "-saved"
+        process_stats_base = run_exec("'python3 1gb_pytorch_correctness.py nn-1gb-base'", jobID_base)
+        process_stats_saved = run_exec("'python3 1gb_pytorch_correctness.py nn-1gb-saved'", jobID_saved)
+        time.sleep(5)
+        run_checkpoint(daemon_pid, jobID_saved, output_dir, process_stats_saved)
+        time.sleep(3) # vary time and  check checkpoints
+        run_restore(daemon_pid, jobID_saved, output_dir)
```

Comment:
are you only running checkpoint/restore here to collect stats?

Reply:
yes, i sometimes comment it out but i can make them more formal functions if you want

Reply:
can add this to the refactoring PR